### PR TITLE
Security issues

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: src/helpers/filesystem.cpp [hunk 1]
```diff
@@ -1,35 +1,97 @@
...
 
-void helpers::copy_directory(const fs::path &src, const fs::path &dest)
+/**
+ * Try to find matching hardlink in hardlinks map. If src is found in the map, dest is filled with corresponding file.
+ * @param hardlinks the hardlinks map (src -> dst)
+ * @param src source path being looked up in hardlinks using equvalent func
+ * @param dest output arg which is filled in case of success
+ * @return true if the hardlink match is found
+ */
+bool find_matching_hadlink(std::map<fs::path, fs::path> &hardlinks, const fs::path &src, fs::path &dest)
```
Comment (Teyras) [inline src/helpers/filesystem.cpp:12]: typo

Comment (krulis-martin) [reply src/helpers/filesystem.cpp:12]: F**k

File: src/sandbox/isolate_sandbox.cpp [hunk 2]
```diff
@@ -160,21 +160,26 @@ void isolate_sandbox::isolate_init_child(int fd_0, int fd_1)
...
+	std::string boxIdArg("--box-id=" + std::to_string(id_));
 
-	args[0] = isolate_binary_.c_str();
-	args[1] = "--cg";
-	args[2] = strdup(("--box-id=" + std::to_string(id_)).c_str());
-	args[3] = "--init";
-	args[4] = NULL;
+	// disk quotas need to be set in initialization
+	auto disk_size_blocks = (limits_.disk_size * 1024) / BLOCK_SIZE;
+	std::string quotaArg("--quota=" + std::to_string(disk_size_blocks) + "," + std::to_string(limits_.disk_files));
```
Comment (Teyras) [inline src/sandbox/isolate_sandbox.cpp:167]: isn't camelCase weird here?

Comment (krulis-martin) [reply src/sandbox/isolate_sandbox.cpp:167]: Sorry, I am so used to it.

**Review by SemaiCZE — APPROVED** [review]

**Pull request merged**